kernel/posix/pthread.c

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+#include <stdio.h>
+#include "ksched.h"
+#include "wait_q.h"
+
+/* Kernel Structure for Number of threads */
+struct k_thread threads[CONFIG_MAX_PTHREAD_COUNT];
+/*
+ *  Default pthread attributes.
+ *  Independent of Implimentation.
+ */
+static pthread_attr_t initPthreadAttrs = {
```

Comment:
Please initialize structures using named initializers; then get rid of comments. It's easy to move stuff around in struct definitions and forget to change the initializers, leading to some really difficult-to-find bugs.